param/version.py

```diff
@@ -233,6 +243,7 @@ def __str__(self):
         """
         if self.release is None: return 'None'
         release = '.'.join(str(el) for el in self.release)
+        release = '%sdev%d' % (release, self.dev) if self.dev else release
```

Comment:
Shouldn't there be a "." before "dev" by PEP440?

Reply:
I think it is fine, this is a snippet Bryan posted:

```python
In [1]: from packaging.version import Version as V

In [2]: V("0.12.10dev3")
Out[2]: <Version('0.12.10.dev3')>

In [3]: v = V("0.12.10dev3")

In [4]: v.is_prerelease
Out[4]: True

In [5]: v.base_version
Out[5]: '0.12.10'

In [6]: v > V("0.12.11")
Out[6]: False
```

That said, I am happy to consider adding the dot - even though our existing dev tags don't have the dot, they have all been lightweight tags so far (and therefore have no effect on ``param.Version``).

@philippjfr Do you want the dot?

Reply:
If we aren't worried about people referencing the old tags (I'm not - they are dev versions!) then I could go back and replace the old dev tags with the new format.

Reply:
Alternatively, we can delete the tags for old dev versions that have been followed by a release. Then we can use the new format going forwards.

Reply:
I'd prefer to leave out the dot personally, bokeh does too.